addons/account/models/account_move.py

```diff
@@ -1369,21 +1369,17 @@ def _compute_access_url(self):
     @api.depends('move_type', 'partner_id', 'company_id')
     def _compute_narration(self):
         use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
-        for move in self:
-            if not move.is_sale_document(include_receipts=True):
-                continue
-            if not use_invoice_terms:
-                move.narration = False
+        invoice_to_update_terms = self.filtered(lambda m: m.is_sale_document(include_receipts=True) and use_invoice_terms)
```

Comment:
small optimization to avoid calling `is_sale_document` when we don't need to
```suggestion
        invoice_to_update_terms = self.filtered(lambda m: use_invoice_terms and m.is_sale_document(include_receipts=True))
```